contracts/StorageContract.sol

```diff
@@ -239,7 +241,6 @@ abstract contract StorageContract is DecentralizedKV {
         MiningLib.update(infos[_shardId], _minedTs, _diff);
 
         require(treasuryReward + minerReward <= address(this).balance, "StorageContract: not enough balance");
-        // TODO: avoid reentrancy attack
```

Comment:
You might want to add a comment explaining that while send() or transfer() prevents reentrancy, it does so at the cost of being incompatible with contracts whose fallback functions require more than 2,300 gas.